### ISSUE TITLE
Potential indexing issue?

### ISSUE THREAD

**Issue by TransGirlCodes** [ngopal/SimpleSmithWatermanCPP#1]
On this line: https://github.com/ngopal/SimpleSmithWatermanCPP/blob/c8e11a12defb0ae7cf4a993b645daf0c1c02b7a0/SmithWaterman.cpp#L93

There is the possibility of indexing into a matrix or string with a j of -1 when the loop does its first iteration, as j is initialized as 0 for the start of the inner loop, should j be set to one to start the inner loop?

**Comment by jianshu93** [comment]
> On this line:
> https://github.com/ngopal/SimpleSmithWatermanCPP/blob/c8e11a12defb0ae7cf4a993b645daf0c1c02b7a0/SmithWaterman.cpp#L93
> There is the possibility of indexing into a matrix or string with a j of -1 when the loop does its first iteration, as j is initialized as 0 for the start of the inner loop, should j be set to one to start the inner loop?

I agree and I try the sequences "CAGTATCGCA" and "AAGTTAGCAG", manually local alignment returns a score of 5 (I believe it is correct) but this script returns 4, I think It is due to the index problem you mentioned.
